network_wrangler/roadway/network.py

```diff
@@ -611,6 +646,242 @@ def delete_nodes(
+    ) -> None:
+        """Splits a link into two links at a specified fraction of its length.
+
+        Args:
+            A: The model_node_id of the start node of the link to split.
+            B: The model_node_id of the end node of the link to split.
+            new_model_node_id: The model_node_id for the new node to be created at the split point.
+            fraction: The fraction along the link's length where the split occurs (0 < fraction < 1).
+            split_reverse_link: If True, also splits the reverse link if it exists. Defaults to True.
+
+        Raises:
+            ValueError: If the link doesn't exist, fraction is invalid, or new_model_node_id already exists.
+        """
+        WranglerLogger.debug(f"split_link() self.links_df.head():\n{self.links_df.head()}")
+        # Find the link with given A and B nodes
+        matching_links = self.links_df[(self.links_df["A"] == A) & (self.links_df["B"] == B)]
+
+        if matching_links.empty:
+            msg = f"Link from node {A} to node {B} not found in network."
+            raise ValueError(msg)
+
+        # If multiple links exist between A and B, use the first one and warn
+        if len(matching_links) > 1:
+            WranglerLogger.warning(
+                f"Multiple links found from node {A} to {B}. Using first link with model_link_id {matching_links.index[0]}."
+            )
+
+        model_link_idx = matching_links.index[0]
+
+        if not 0 < fraction < 1:
+            msg = f"Fraction must be between 0 and 1 (exclusive), got {fraction}."
+            raise ValueError(msg)
+
+        if new_model_node_id in self.nodes_df.index:
+            msg = f"Node with model_node_id {new_model_node_id} already exists in network."
+            raise ValueError(msg)
+
+        # Get the original link
+        orig_link = self.links_df.loc[model_link_idx].copy()
+        WranglerLogger.debug(f"Splitting link:\n{orig_link}")
+
+        # Get the geometry to split (use shape geometry if available)
+        if pd.notna(orig_link.get("shape_id")) and orig_link["shape_id"] in self.shapes_df.index:
+            geometry = self.shapes_df.loc[orig_link["shape_id"], "geometry"]
+        else:
+            geometry = orig_link["geometry"]
+
+        # Calculate the split point
+        split_point = geometry.interpolate(fraction, normalized=True)
+
+        # Create the new node
+        # The attribute osm_node_id is not required so don't set it
+        new_node_data = {
+            "model_node_id": new_model_node_id,
+            "X": split_point.x,
+            "Y": split_point.y,
+        }
+
+        # Copy additional attributes from node A if they exist
+        node_a_data = self.nodes_df.loc[self.nodes_df["model_node_id"] == orig_link["A"]].iloc[0]
+        # Copy all columns except geometry, X, Y, and model_node_id
+        for col in self.nodes_df.columns:
+            if (
+                col not in ["model_node_id", "X", "Y", "geometry", "osm_node_id"]
```

Comment:
Use params in [NodesTable definitions](https://github.com/network-wrangler/network_wrangler/blob/main/network_wrangler/models/roadway/tables.py#L330) for this list of vars.